website/docs/topics/long_contexts.md

```diff
@@ -0,0 +1,354 @@
+
+#### Example 2: Limiting the Number of Tokens
+
+To adhere to token limitations, use the `MessageTokenLimiter` transformation. This limits tokens per message and the total token count across all messages:
+
+```python
+# Limit the token limit per message to 3 tokens
+token_limit_transform = transforms.MessageTokenLimiter(max_tokens_per_message=3)
+
+processed_messages = token_limit_transform.apply_transform(copy.deepcopy(messages))
+
+pprint.pprint(processed_messages)
+```
+
+```console
+[{'content': 'hello', 'role': 'user'},
+{'content': [{'text': 'there', 'type': 'text'}], 'role': 'assistant'},
+{'content': 'how', 'role': 'user'},
+{'content': [{'text': 'are you doing', 'type': 'text'}], 'role': 'assistant'},
+{'content': 'very very very', 'role': 'user'}]
+```
+
+We can see that we were able to limit the number of tokens to 3, which is equivalent to 3 words for this instance.
+
+### Apply Transformations Using Agents
+
+So far, we have only tested the `MessageHistoryLimiter` and `MessageTokenLimiter` transformations individually, let's test these transformations with AutoGen's agents.
+
+#### Setting Up the Stage
+
+```python
+import os
+import copy
+
+import autogen
+from autogen.agentchat.contrib.capabilities import transform_messages, transforms
+from typing import Dict, List
+
+config_list = [{"model": "gpt-3.5-turbo", "api_key": os.getenv("OPENAI_API_KEY")}]
+
+# Define your agent; the user proxy and an assistant
+assistant = autogen.AssistantAgent(
+    "assistant",
+    llm_config={"config_list": config_list},
+)
+user_proxy = autogen.UserProxyAgent(
+    "user_proxy",
+    human_input_mode="NEVER",
+    is_termination_msg=lambda x: "TERMINATE" in x.get("content", ""),
+    max_consecutive_auto_reply=10,
+)
+```
+
+```{=mdx}
+:::tip
+Learn more about configuring LLMs for agents [here](/docs/topics/llm_configuration).
+:::
+```
+
+We first need to write the `test` function that creates a very long chat history by exchanging messages between an assistant and a user proxy agent, and then attempts to initiate a new chat without clearing the history, potentially triggering an error due to token limits.
+
+```python
+# Create a very long chat history that is bound to cause a crash for gpt 3.5
+def test(assistant: autogen.ConversableAgent, user_proxy: autogen.UserProxyAgent):
+    for _ in range(1000):
+        # define a fake, very long messages
+        assitant_msg = {"role": "assistant", "content": "test " * 1000}
+        user_msg = {"role": "user", "content": ""}
+
+        assistant.send(assitant_msg, user_proxy, request_reply=False, silent=True)
+        user_proxy.send(user_msg, assistant, request_reply=False, silent=True)
+
+    try:
+        user_proxy.initiate_chat(assistant, message="plot and save a graph of x^2 from -10 to 10", clear_history=False)
+    except Exception as e:
+        print("Encountered an error with the base assistant")
+        print(e)
+        print("\n\n")
```

Comment:
`print(f"Encountered an error with the base assistant. \n{e}"}`
probably cleaner, to avoid 2 extra lines of code.